Source/WebCore/loader/NavigationScheduler.cpp

```diff
@@ -386,9 +386,8 @@ class ScheduledHistoryNavigationByKey : public ScheduledNavigation {
 
         auto completionHandler = std::exchange(m_completionHandler, nullptr);
 
-        Ref rootFrame = localFrame->rootFrame();
         RefPtr upcomingTraverseMethodTracker = localFrame->window()->navigation().upcomingTraverseMethodTracker(m_key);
-        page->goToItemForNavigationAPI(rootFrame, *historyItem, FrameLoadType::IndexedBackForward, *localFrame, upcomingTraverseMethodTracker.get());
+        page->goToItemForNavigationAPI(*localFrame, *historyItem, FrameLoadType::IndexedBackForward, *localFrame, upcomingTraverseMethodTracker.get());
```

Comment:
If we're making this change, we should also change the frame that is passed to `recursiveGatherFramesToNavigate` in `HistoryController::goToItemForNavigationAPI`

Reply:
Done, I also updated the `targetItem` and `fromItem` to be from that frame. Hopefully the bots are happy with that.